File: hephaestus/src/__tests__/fakes.py
import json

PAGE_ID = "test-id"
pageForTest = lambda schemas: {
    "id": PAGE_ID,
    "body": f"""
        <html>
          <head>
            {"".join([f'<script type="application/ld+json">{json.dumps(schema)}</script>' for schema in schemas])}
            <script src=\"_ascript\"></script>
          </head>
        </html>
    """,
}

File: packages/indexer/__tests__/utils/fakes.ts
import { PAGE } from "@tissai/db/mocks"

export { SITE, PAGE, BRAND, PRODUCT, SELLER, OFFER } from "@tissai/db/mocks"

export const pageWithSchema = (...schemas: object[]) => ({
  ...PAGE,
  body: `
    <html>
      <head>
        ${schemas
          .map(
            (schema) => `
          <script type="application/ld+json">
            ${JSON.stringify(schema)}
          </script>
        `,
          )
          .join()}
      </head>
    </html>
  `,
})
